### DISCUSSION TITLE
Continuous Deployment

### DISCUSSION THREAD

**Discussion by LecrisUT** [Homebrew/discussions#4514] [Tap maintenance and Homebrew development]
### Description of issue

Are there any work to make an official homebrew GithubAction and equivalents so that it can be implemented in other project's CD?

**Comment by p-linnane** [comment]
I'm not 100% sure what you're asking. If you're referring to being able to bump the formula as part of your pipeline, you can use this: https://github.com/marketplace/actions/bump-homebrew-formula

**Reply by LecrisUT** [reply]
Indeed that is about what I was looking for. There are a few issues with it not being an official homebrew GH action like requiring a `COMMITTER_TOKEN` instead of using a specific github bot app as the committer. But maybe I don't quite understand how it is supposed to work.

**Reply by carlocab** [reply]
The difficulty is that the action needs to be able to write to a fork of Homebrew/homebrew-core, which AFAIK requires a PAT.

**Reply by LecrisUT** [reply]
That's why it should be paired with GH app. The GH app provides a bot user that can be used as the author for the commit. That is then installed on the fork to give it push permissions. I've mostly seen the implementation where the GH app is an external service, but maybe it can be manipulated within the github action.

**Reply by carlocab** [reply]
Based on [these docs](https://docs.github.com/en/apps/creating-github-apps/guides/making-authenticated-api-requests-with-a-github-app-in-a-github-actions-workflow), it looks like you need to create your own GitHub app to do this.

The author of the GitHub action referenced above works for GitHub, so I'm guessing that if there were a way to make a third-party app (with respect to your repo/org) do what you're looking for, they'd have made one already.
